patch/patch-runelite/src/main/kotlin/net/rsprox/patch/runelite/RuneLitePatcher.kt

```diff
@@ -14,46 +16,98 @@ import java.security.KeyStore
 import java.security.KeyStore.PasswordProtection
 import java.security.KeyStore.PrivateKeyEntry
 import java.security.MessageDigest
-import kotlin.io.path.ExperimentalPathApi
-import kotlin.io.path.Path
-import kotlin.io.path.copyTo
-import kotlin.io.path.deleteIfExists
-import kotlin.io.path.deleteRecursively
-import kotlin.io.path.exists
-import kotlin.io.path.extension
-import kotlin.io.path.isRegularFile
-import kotlin.io.path.moveTo
-import kotlin.io.path.nameWithoutExtension
-import kotlin.io.path.readBytes
-import kotlin.io.path.readText
-import kotlin.io.path.writeBytes
-import kotlin.io.path.writeText
+import kotlin.io.path.*
 
 @Suppress("DuplicatedCode", "SameParameterValue")
-public class RuneLitePatcher : Patcher<Unit> {
-    public fun patch(
+public class RuneLitePatcher : Patcher<RuneLitePatchCriteria> {
+    override fun patch(
         path: Path,
-        rsa: String,
-        port: Int,
+        criteria: RuneLitePatchCriteria,
     ): PatchResult {
-        return patch(
-            path,
-            rsa,
-            "",
-            "",
-            port,
-            Unit,
+        logger.debug { "Running launcher to build jar file." }
+
+        if (criteria.bootstrap == null || criteria.artifactsDir == null) {
+            throw IllegalArgumentException("Bootstrap and artifacts directory must be provided.")
+        }
+
+        if (criteria.rsa == null) {
+            throw IllegalArgumentException("RSA modulus must be provided.")
+        }
+
+        val timestamp = System.currentTimeMillis()
+        val socketFile = Path.of(System.getProperty("user.home")).resolve(".rsprox").resolve("sockets").resolve("$timestamp.socket").toFile()
+        val socket = AFUNIXServerSocket.newInstance()
+        logger.debug { "Binding an AF UNIX Socket to ${socketFile.name}" }
+        socket.bind(AFUNIXSocketAddress.of(socketFile))
+
+        val launcher = RuneLiteLauncher(criteria.bootstrap, criteria.artifactsDir)
+
+        val args = launcher.getLaunchArgs(
+            port = criteria.port,
+            rsa = criteria.rsa,
+            javConfig = criteria.javConfigUrl,
+            socket = timestamp.toString(),
+        )
+
+        val builder = ProcessBuilder()
+            .inheritIO()
+            .command(args)
+
+        val process = builder.start()
+
+        val channel = socket.accept()
+        val output = channel.outputStream
+        output.write("old-rsa-modulus:".encodeToByteArray())
+        output.flush()
+        socket.close()
+
+        // Wait for the original Launcher patcher process to finish
+        while (process.isAlive) {
+            Thread.sleep(50)
+        }
```

Comment:
I believe this could simply be replaced with process.waitFor(), based on the documentation:
> Causes the current thread to wait, if necessary, until the
process represented by this {@ code Process} object has
terminated.  This method returns immediately if the process
has already terminated.  If the process has not yet
terminated, the calling thread will be blocked until the
process exits.